internal/pkg/handler/upgrade.go

```diff
@@ -154,7 +154,7 @@ func PerformRollingUpgrade(clients kube.Clients, config util.Config, upgradeFunc
 		}
 		result := constants.NotUpdated
 		reloaderEnabled, err := strconv.ParseBool(reloaderEnabledValue)
-		if err == nil && reloaderEnabled {
+		if (err != nil && options.AutoReloadAll) || (err == nil && (reloaderEnabled || options.AutoReloadAll)) {
```

Comment:
In order to behave as documented, I think you need to omit `|| options.AutoReloadAll` here, so that it will not reload when the auto annotation is explicitly set to `"false"`.

Reply:
Thanks for your feedback. You're right, and the condition is false. Notice that also your suggestion isn't enough, since it'll still reload when `err != nil && options.AutoReloadAll`, which isn't a desired behavior, since all errors should be treated as `false`.

To ensure this time I didn't make a mistake, I created a truth table and inferred the correct condition using a Karnaugh Map.
<img width="529" alt="image" src="https://github.com/stakater/Reloader/assets/5053052/d73213ee-7b3f-4e1a-9c8c-13f17b5951dd">

Reply:
It was quite confusing, but I realized that there was an error in the Karnaugh Map I created. I didn't think of the case that `reloaderEnabled` is unset, causing `err` to be non-nil.
I just fixed my table and realized that the condition we want to check is:
`reloaderEnabled || reloaderEnabledValue == "" && options.AutoReloadAll`

Here's the truth table I wrote and the map I created from it:
![image](https://github.com/stakater/Reloader/assets/5053052/d5dbc368-bb10-4a4c-883c-5a4624918a17)